### PR TITLE
Version manifest

### PR DESCRIPTION
**Pull request by wlandau**
This PR adds `record_versions()` which creates and updates a record (`versions.json`) of package versions and the highest version recorded so far. It also keeps track of the MD5 hashes, which I assume are guaranteed to update whenever the package source changes (currently looking for confirmation on that). `record_versions()` also writes `version_issues.json` which is a subset of the manifest containing packages for which either:

1. The current package release does not have the highest version number ever released for the package.
2. The package is at its highest version number, but that same version was previously released under a different MD5.

Supports https://github.com/r-releases/help/issues/21.

### CODE REVIEW

**Comment by wlandau** [comment]
`version_issues.json` is written as its own file in order to help with https://github.com/r-releases/help/issues/6 (cc @llrs). Only a small number of packages should have version issues, so a new `releases::check_releases()` function (or whatever we decide to call it) should be able to download it quickly. No need to support a whole web API just to scrape parts of the full `versions.json`.

**Comment by wlandau** [comment]
https://github.com/r-releases/help/issues/21#issuecomment-1978467899 is really helpful. If those MD5s are brittle, then I agree we shouldn't use them here. Trouble is I don't know how to get the GitHub SHAs from the R-universe API. @shikokuchuo, do you know how?

**Comment by wlandau** [comment]
Okay, so I added a `check_hash` argument just now. `check_hash = FALSE` allows us to do everything we can using the versions alone. This should unblock https://github.com/r-releases/help/issues/6 from initial development while we think of something better for hashes..

File: R/record_versions.R
Comment (shikokuchuo) [inline R/record_versions.R:52]: Should swap lines 51 and 52 for efficiency.

File: R/record_versions.R
Comment (shikokuchuo) [inline R/record_versions.R:63]: is this necessary? if so: `lapply(out, as.character)` before line 60 do.call vec_rbind ?

File: R/record_versions.R
Comment (shikokuchuo) [inline R/record_versions.R:77]: Can we change to == 1L pls. For maintainability - I had to look up function documentation for utils::compareVersion to check this.

**Review by shikokuchuo — APPROVED** [review]

File: R/record_versions.R
Comment (wlandau) [reply R/record_versions.R]: Good idea, done in 2c39169ba7e27746fb317f5099d06af10f67c2fd

File: R/record_versions.R
Comment (wlandau) [reply R/record_versions.R]: It is necessary because otherwise those columns are lists. Shortened in 6addffaeb595313504a299a2ba59e2b437f33e97.

File: R/record_versions.R
Comment (wlandau) [reply R/record_versions.R]: I get paranoid about floating point in R, but sure, implemented in 455de781dceefa6c17a1d56a5eb8386ae777ba5b

Comment (shikokuchuo) [reply R/record_versions.R]: oh you're right this is so confusing. The documentation says "0 if the numbers are equal, -1 if b is later and 1 if a is later (analogous to the C function strcmp)." It is NOT analogous if the function returns a double rather than an int!

File: R/record_versions.R
Comment (shikokuchuo) [reply R/record_versions.R:25]: I'll leave it to you. But even if the package is not attached, the search path should still always start with the package namespace shouldn't it?

**Pull request merged**
